src/main/java/de/tum/in/www1/artemis/web/rest/lecture/OnlineUnitResource.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+    /**
+     * PUT /lectures/:lectureId/online-units : Updates an existing online unit .
+     *
+     * @param lectureId      the id of the lecture to which the online unit belongs to update
+     * @param onlineUnit the online unit to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated onlineUnit
+     */
+    @PutMapping("/lectures/{lectureId}/online-units")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<OnlineUnit> updateOnlineUnit(@PathVariable Long lectureId, @RequestBody OnlineUnit onlineUnit) {
+        log.debug("REST request to update an online unit : {}", onlineUnit);
+        if (onlineUnit.getId() == null) {
+            throw new BadRequestException();
+        }
+
+        if (onlineUnit.getLecture() == null || onlineUnit.getLecture().getCourse() == null) {
+            throw new ConflictException("Lecture unit must be associated to a lecture of a course", "onlineUnit", "lectureOrCourseMissing");
+        }
+
+        // Validate the URL
+        try {
+            new URL(onlineUnit.getSource());
+        }
+        catch (MalformedURLException exception) {
+            throw new BadRequestException();
+        }
+
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, onlineUnit.getLecture().getCourse(), null);
+
+        if (!onlineUnit.getLecture().getId().equals(lectureId)) {
+            throw new ConflictException("Requested lecture unit is not part of the specified lecture", "onlineUnit", "lectureIdMismatch");
+        }
+
+        OnlineUnit result = onlineUnitRepository.save(onlineUnit);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, onlineUnit.getId().toString())).body(result);
+    }
+
+    /**
+     * POST /lectures/:lectureId/online-units : creates a new online unit.
+     *
+     * @param lectureId      the id of the lecture to which the online unit should be added
+     * @param onlineUnit the online unit that should be created
+     * @return the ResponseEntity with status 201 (Created) and with body the new online unit
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/lectures/{lectureId}/online-units")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<OnlineUnit> createOnlineUnit(@PathVariable Long lectureId, @RequestBody final OnlineUnit onlineUnit) throws URISyntaxException {
+        log.debug("REST request to create onlineUnit : {}", onlineUnit);
+        if (onlineUnit.getId() != null) {
+            throw new BadRequestException();
+        }
+
+        // Validate the URL
+        try {
+            new URL(onlineUnit.getSource());
+        }
+        catch (MalformedURLException exception) {
+            throw new BadRequestException();
+        }
+
+        Lecture lecture = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoalsElseThrow(lectureId);
```

Comment:
Why do we fetch the posts and learning goals here? Are they crucial for this operation?

Reply:
Thanks for pointing that out, I will fix it with some more refactoring in another branch.